lib/Migration/Version2801Date202309200001.php

```diff
@@ -86,8 +86,10 @@ public function postSchemaChange(IOutput $output, Closure $schemaClosure, array
 					$delete->executeStatement();
 				}
 			}
+			if ($result !== null) {
```

Comment:
this null check seems unnecessary

Reply:
This null check is needed and is what fixes bugs #813 and #819. i.e. 

```Error: Call to a member function closeCursor() on null in /var/www/nextcloud/apps/fulltextsearch/lib/Migration/Version2801Date202309200001.php:91```

So it's quite necessary :D

Reply:
yes but this is already resolved by moving it into the inner scope.

If it is null the code would crash here (so it would never get to your check):
https://github.com/nextcloud/fulltextsearch/blob/cb8794fef8e7268812e337cb7a976eb81756f223/lib/Migration/Version2801Date202309200001.php#L66
So `$result !== null`  should always be true.

Reply:
You are right. I have updated the PR to remove this check. Hopefully this code is good to go now! :)